### ISSUE TITLE
Update dependency constraint to allow for liquid v5.4.0

### ISSUE THREAD

**Issue by jekyllbot** [jekyll/jekyll#9094]
Hey there! :wave:

I noticed that the constraint you have for liquid doesn't allow for the latest version to be used.

The constraint I found was `~> 4.0`, and the latest version available is `5.4.0`.

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

**Comment by HoneyryderChuck** [comment]
As per [this changelog entry](https://github.com/Shopify/liquid/blob/master/History.md#500--2021-01-06), jekyll will need to update to liquid 5 to work with ruby 3.2 .

**Comment by yboulkaid** [comment]
This is blocked by https://github.com/Shopify/liquid/issues/1566

**Comment by ashmaroli** [comment]
Yes, that issue is a wrench in our mechanism but apart from that, I had unknowingly introduced a coupling to Liquid internals:

https://github.com/jekyll/jekyll/blob/2cc51e6abab049621f49e3375a138c296a0494d5/lib/jekyll/filters.rb#L400-L419

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least two months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the latest <code>3.x-stable</code> or <code>master</code>/<code>main</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, please consider building it first as a plugin. Jekyll 3 introduced [hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be built as a plugin, then please provide more information about why in order to keep this issue open.

This issue will automatically be closed in two months if no further activity occurs. Thank you for all your contributions.

**Comment by parkr** [comment]
Cc https://github.com/github/pages-gem/issues/859

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least two months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the latest <code>3.x-stable</code> or <code>master</code>/<code>main</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, please consider building it first as a plugin. Jekyll 3 introduced [hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be built as a plugin, then please provide more information about why in order to keep this issue open.

This issue will automatically be closed in two months if no further activity occurs. Thank you for all your contributions.

**Comment by parkr** [comment]
This is still blocked by https://github.com/Shopify/liquid/issues/1566.

**Comment by ashmaroli** [comment]
@parkr The issue you have linked to isn't the only blocker.
There's also the issue described in https://github.com/Shopify/liquid/pull/1663 and https://github.com/Shopify/liquid/pull/1662

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least two months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the latest <code>3.x-stable</code> or <code>master</code>/<code>main</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, please consider building it first as a plugin. Jekyll 3 introduced [hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be built as a plugin, then please provide more information about why in order to keep this issue open.

This issue will automatically be closed in two months if no further activity occurs. Thank you for all your contributions.

**Comment by nickmccurdy** [comment]
Duplicate of #8535
